### PR TITLE
Source extensions properly, fixes #148192

### LINKED ISSUE #148192: Settings editor funnel button: How does it source extensions?
Testing #147935

- Open Settings editor, select funnel -> `Extension ID...`
- See several extension suggestions:

image

- I want to view Remote-Containers settings, so I start searching `containers`
- No results, so I'd need to find and paste the Remote-Containers ID myself

image

How do these suggestions get populated? I thought it would've been all extensions installed, or all installed extensions with settings, but that doesn't appear to be the case (in the latter scenario, there are extensions populated that don't have any settings to show).

### PR DESCRIPTION
**Pull request by rzhao271**
This PR fixes #148192

When a user presses the filter button in the Settings editor search widget and selects the "Extensions" option, it used to be that we would list all built-in extensions, even ones that didn't provide any settings. Another problem was we would not list any user-installed extensions. Now, we list built-in extensions and user-installed extensions, but only ones that contribute settings (i.e. have a contributes.configuration section).

### CODE REVIEW

File: src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts [hunk 1]
```diff
@@ -275,8 +275,15 @@ export class SettingsEditor2 extends EditorPane {
 			SettingsEditor2.SUGGESTIONS.push(`@${LANGUAGE_SETTING_TAG}`);
 		}
 
-		extensionManagementService.getInstalled(ExtensionType.System).then(extensions => {
-			this.installedExtensionIds = extensions.map(extension => extension.identifier.id);
+		Promise.all([extensionManagementService.getInstalled(ExtensionType.System), extensionManagementService.getInstalled(ExtensionType.User)]).then(extensions => {
```
Comment (sandy081) [inline src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts:278]: You can get all extension without passing any type, eg: `extensionManagementService.getInstalled()`

**Review by sandy081 — APPROVED** [review]
LGTM (except for the change I mentioned).

File: src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts (hunk 1 above)
Comment (rzhao271) [reply src/vs/workbench/contrib/preferences/browser/settingsEditor2.ts]: That greatly simplified the implementation. Added a commit to the PR.

**Comment by roblourens2** [comment]
LGTM

**Pull request merged**